pontoon/sync/core/translations_from_repo.py

```diff
@@ -0,0 +1,495 @@
+                    tx.approved = True
+                    tx.approved_date = now
+                new_translations.append(tx)
+                actions.append(
+                    ActionLog(
+                        action_type=ActionLog.ActionType.TRANSLATION_CREATED,
+                        created_at=now,
+                        performed_by=log_user,
+                        translation=tx,
+                    )
+                )
+                translations_to_reject |= Q(
+                    entity_id=entity_id, locale_id=locale_id, plural_form=plural_form
+                )
+
+    if translations_to_reject:
+        rejected = Translation.objects.filter(rejected=False).filter(
+            translations_to_reject
+        )
+        actions.extend(
+            ActionLog(
+                action_type=ActionLog.ActionType.TRANSLATION_REJECTED,
+                created_at=now,
+                performed_by=log_user,
+                translation=tx,
+                is_implicit_action=True,
+            )
+            for tx in rejected
+        )
+        reject_count = rejected.update(
+            active=False,
+            approved=False,
+            approved_user=None,
+            approved_date=None,
+            rejected=True,
+            rejected_user=None,
+            rejected_date=now,
+            pretranslated=False,
+            fuzzy=False,
+        )
+        if reject_count:
+            TranslationMemoryEntry.objects.filter(
+                translation__in=[tx.pk for tx in rejected]
+            ).delete()
+            log.info(
+                f"[{project.slug}] Rejected {str_n_translations(reject_count)} from repo changes"
+            )
+
+    update_count = (
+        Translation.objects.bulk_update(suggestions.values(), list(update_fields))
+        if update_fields
+        else 0
+    )
+    if update_count:
+        count = (
+            str(approve_count)
+            if approve_count == update_count
+            else f"{approve_count}/{update_count}"
+        )
+        log.info(
+            f"[{project.slug}] Approved {str_n_translations(count)} from repo changes"
+        )
+
+    created = Translation.objects.bulk_create(new_translations)
+    if created:
+        log.info(
+            f"[{project.slug}] Created {str_n_translations(created)} from repo changes"
+        )
+
+    if actions:
+        ActionLog.objects.bulk_create(actions)
+
+    return created, list(suggestions.values())
+
+
+def str_n_translations(n: int | Sized) -> str:
+    if not isinstance(n, int):
+        n = len(n)
+    return "1 translation" if n == 1 else f"{n} translations"
+
+
+def get_path_locale(path_vars: dict[str, str]) -> str | None:
+    if "locale" in path_vars:
+        return path_vars["locale"]
+    elif "android_locale" in path_vars:
+        return parse_android_locale(path_vars["android_locale"])
+    else:
+        return None
+
+
+def add_errors(translations: list[Translation]) -> None:
```

Comment:
It could be warnings, too. Maybe this?

```suggestion
def add_failed_checks(translations: list[Translation]) -> None:
```